crates/common/batcher/src/batcher.rs

```diff
@@ -108,13 +108,9 @@ impl<B: Batchable> Batcher<B> {
     }
 
     fn find_target_batch(&mut self, event_time: OffsetDateTime) -> Option<&mut Batch<B>> {
-        for batch in &mut self.batches {
-            if batch.batch_start() <= event_time && event_time <= batch.batch_end() {
-                return Some(batch);
-            }
-        }
-
-        None
+        self.batches
+            .iter_mut()
+            .find(|batch| batch.batch_start() <= event_time && event_time <= batch.batch_end())
```

Comment:
Okay. However, one is not better than the other. And this is why I have a mix feeling to treat clippy warnings as errors.